### ISSUE TITLE
Is there a way to copy the included picture to anki?

### ISSUE THREAD

**Issue by panmengguan** [eyeinsky/org-anki#17]
Sometimes, we hope to synchronize some org-mode entries which have included a picture. For example:

`[[XXX.png]]`

And we hope this picture could copy to the anki media folder and after =org-anki-sync-entry= is executed, we could see this picture in Anki. Is there any way to do this ?

**Comment by eyeinsky** [comment]
Not at the moment, but when the other features are done I will look into this (and pull requests are also welcome!).

**Comment by ingtshan** [comment]
> we could see this picture in Anki. Is there any way to do this ?

Current time, there is a way to view picture in anki bypass synchronizing media, but you should upload your picture with tool like [gee1k/uPic](https://github.com/gee1k/uPic) first, and use picture url at org-mode entries. 

Consider as a temporary method.

**Comment by eyeinsky** [comment]
Linking PR https://github.com/eyeinsky/org-anki/pull/43 to sync images, perhaps someone would like a say there.

**Comment by IsakFalk** [comment]
@ingtshan

Could you elaborate on how you made this work using URL?

**Comment by eyeinsky** [comment]
By reading what was said, then adding something like this `@@html:<img src="https://url-to-your-image">@@` in org-mode and then hosting the images somewhere should work.

**Comment by v-Nyo** [comment]
I really hope this will be implemented someday. This is the only reason I am the anki-editor fork.
Also not just for pictures but media in general (audio, GIF , videos..).

**Comment by eyeinsky** [comment]
I wonder what to do with images which have the same name. I.e, if org entries have a link to ~/picture.jpg and ~/subfolder/picture.jpg, then moving those into the anki media folder will overwrite them. I'm thinking of either hashing them and prepending the hash.

**Comment by v-Nyo** [comment]
sounds good. But does it have to be a hash of the file? What if you want to add your 1TB mp4 to your anki card?

I think appending/renaming with a random number/uuid would be better.
https://github.com/emacsmirror/uuidgen

Not sure how it should handle updating that card? Where would you store/get the uuid/filehash? 
I don't think it's desirable to use anki as file versioning when you update the image. Even tho anki does a really good job of cleaning orphaned media.

**Comment by eyeinsky** [comment]
Right, I think the code for it will be isolated enough to switch between methods to find a fitting one!

While thinking about it though, perhaps other methods to differentiate files with the same filename could include:
- using file's size and prepending it to the name
- then comparing content on-demand (until first difference)

And yes, cleaning orphaned media could be ran now and then to remove unused files.

**Comment by eyeinsky** [comment]
Please test the latest version and customize the Emacs variable `org-anki-upload-images` to `t`, this will upload all `[[file:...]]` links to Anki. If that could be improved then make another issue. :)
